docs/sources/staged.rst

```diff
@@ -196,6 +197,19 @@ Files in this directory must have metadata included in ``staged.yaml``.
 
 Will yield instances of :class:`~pushsource.AmiPushItem`.
 
+root/destination/CLOUD_IMAGES/\*
```

Comment:
What's implemented doesn't match what's in the design doc.

Design doc: https://url.corp.redhat.com/ab

The idea was there would be dirs like `root/<destination>/CLOUD_IMAGES/<build>`, where each "build" dir contains a "resources.yaml" and referenced images. 

In practice, there would always only be one destination - starmap - while there would be multiple builds.

For instance the doc had: "Example of a valid staged directory with AmiPushItems for RHEL 9.0, 9.4 and 9.6":

```
/starmap/CLOUD_IMAGES/
  /rhel-ec2-9.0-20241010.1/
    resources.yaml
    rhel-ec2-9.0-20241010.1.x86_64.raw.xz
  /rhel-ec2-9.4-20241010.1/
    resources.yaml
    rhel-ec2-9.4-20241010.1.x86_64.raw.xz
  /rhel-ec2-9.6-20241010.1/
    resources.yaml
    rhel-ec2-9.6-20241010.1.x86_64.raw.xz
```

What you've implemented instead seems to be this:

- dirs `root/<destination>/CLOUD_IMAGES/` exist
- the CLOUD_IMAGES directory itself contains `resources.yaml` and the images (no further subdir)
- In order to include multiple `resources.yaml`, you can have multiple different destinations.  For instance in your test data you used `build1`, `build2`, `build3` as some destinations.

Is it intentionally different from the design?  I think the agreed-upon design in the doc was better.

Reply:
I've reworked this to match the agreed design in the doc. I did have to include an empty staged.yaml file otherwise it would be skipped in the scan.